app/models/portfolio_item.rb

```diff
@@ -24,4 +27,12 @@ def resolved_workflow_refs
   def item_workflow_ref
     [self, portfolio].detect(&:workflow_ref)&.workflow_ref
   end
+
+  def validate_workflow
+    raise Catalog::InvalidParameter if workflow_ref.blank?
+
+    Approval::Service.call(ApprovalApiClient::WorkflowApi) { |api| api.show_workflow(workflow_ref) }
+  rescue Catalog::ApprovalError
+    raise Catalog::InvalidParameter, "Invalid workflow_ref"
```

Comment:
@lindgrenj6 
If the Approval Service is down would we be overwriting that error with InvalidParameter? Shouldn't the Approval API be sending a 404 not found that we should pick up and forward to our caller.

Reply:
We will be discussing this Thursday. It hasn't been decided yet if we're going to be validating external references such as approval workflows (this change also is spill over from #340, once that is merged this won't be here. Depending how the discussion goes it may just get pulled out completely).

Reply:
So after discussing this morning, I'm going to leave this as-is, but in the future we're going to take the error message from this and include it in the error response.
(this change is also only in #340, I just built this branch off of that one for the refactor).